app/models/item.rb

```diff
@@ -1,3 +1,13 @@
 class Item < ApplicationRecord
   belongs_to :supermarket
+  has_many :customer_items
+  has_many :customers, through: :customer_items
+
+  def supermarket_name_by_item
+    self.supermarket.name
```

Comment:
Nice readable name for this method! It's also okay to do `item.supermarket.name` in the view - reaching into an association's attributes is not an MVC infraction (but pulling it into a model method isn't bad! Just a preference!)